be/src/runtime/datetime_value.cpp

```diff
@@ -1557,13 +1557,28 @@ bool DateTimeValue::from_unixtime(int64_t timestamp, const std::string& timezone
     if (timestamp < 0 || timestamp > 253402271999L) {
         return false;
     }
+
     boost::local_time::time_zone_ptr local_time_zone = TimezoneDatabase::find_timezone(timezone);
     if (local_time_zone == nullptr) {
         return false;                            
     }
-    boost::local_time::local_date_time lt(boost::posix_time::from_time_t(timestamp), local_time_zone);
+
+    int64_t current_t = timestamp;
+    boost::posix_time::ptime time = boost::posix_time::ptime(boost::gregorian::date(1970,1,1));
+
+    while(current_t > 0) {
```

Comment:
```suggestion
    while (current_t > 0) {
```
Add some comments about what this block is doing.